staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go

```diff
@@ -66,19 +68,50 @@ func newService(name string, uid types.UID, serviceType v1.ServiceType) *v1.Serv
 			Type: serviceType,
 		},
 	}
+	for _, tw := range tweaks {
+		tw(s)
+	}
+	return s
 }
 
-func newETPLocalService(name string, serviceType v1.ServiceType) *v1.Service {
-	return &v1.Service{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      name,
-			Namespace: "default",
-			UID:       "777",
-		},
-		Spec: v1.ServiceSpec{
-			Type:                  serviceType,
-			ExternalTrafficPolicy: v1.ServiceExternalTrafficPolicyLocal,
-		},
+func tweakAddETP(etpType v1.ServiceExternalTrafficPolicyType) serviceTweak {
+	return func(s *v1.Service) {
+		s.Spec.ExternalTrafficPolicy = etpType
+	}
+}
+
+func tweakAddLBI(ip string) serviceTweak {
+	return func(s *v1.Service) {
+		s.Status.LoadBalancer.Ingress = []v1.LoadBalancerIngress{{IP: ip}}
+	}
+}
+
+func tweakAddPorts(protocol v1.Protocol, targetPort string) serviceTweak {
```

Comment:
We have a lot of this for Service but it's all implemented against v1.Service.  You can copy some logic from there.

pkg/api/service/testing/make.go